$NetBSD: patch-Makefile,v 1.3 2024/03/10 21:02:21 vins Exp $

* Honour pkgsrc compiler/linker flags.
* Do not require pkg-config.
* Provide a pkgsrc-compliant install target.

--- Makefile.orig	2023-01-27 20:35:24.000000000 +0000
+++ Makefile
@@ -4,14 +4,11 @@ VERSION!=	[ -d .git ] && \
 		echo "git-`git rev-list --abbrev-commit --tags --max-count=1`" || \
 		echo "${VERSION}"
 
-CC?=		cc
-PREFIX?=	/usr/local
-PKGLIBS=	x11 xft xrandr xtst xres
-CFLAGS+=	-O2 -Wall \
-		-Wunused -Wmissing-prototypes -Wstrict-prototypes \
-		`pkg-config --cflags ${PKGLIBS}` \
+CC?=		@CC@
+PREFIX?=	@PREFIX@
+CFLAGS+=	-Wall -Wunused -Wmissing-prototypes -Wstrict-prototypes \
 		-DVERSION=\"${VERSION}\"
-LDFLAGS+=	`pkg-config --libs ${PKGLIBS}`
+LIBS+=		-lX11 -lXft -lXrandr -lXtst -lXRes
 
 # uncomment to enable debugging
 #CFLAGS+=	-g -DDEBUG=1
@@ -20,7 +17,7 @@ LDFLAGS+=	`pkg-config --libs ${PKGLIBS}`
 #CFLAGS+=	-DSENDCMD_DEBUG=1
 
 BINDIR=		${DESTDIR}$(PREFIX)/bin
-MANDIR=		${DESTDIR}$(PREFIX)/man/man1
+MANDIR=		${DESTDIR}$(PREFIX)/@PKGMANDIR@/man1
 
 SRC!=		ls *.c
 OBJ=		${SRC:.c=.o}
@@ -31,12 +28,13 @@ MAN=		sdorfehs.1
 all: sdorfehs
 
 sdorfehs: $(OBJ)
-	$(CC) -o $@ $(OBJ) $(LDFLAGS)
+	$(CC) -o $@ $(OBJ) $(LDFLAGS) $(LIBS)
 
 install: all
-	mkdir -p $(BINDIR) $(MANDIR)
-	install -s $(BIN) $(BINDIR)
-	install -m 644 $(MAN) $(MANDIR)
+	$(BSD_INSTALL_PROGRAM_DIR) $(BINDIR)
+	$(BSD_INSTALL_PROGRAM) $(BIN) $(BINDIR)
+	$(BSD_INSTALL_MAN_DIR) $(MANDIR)
+	$(BSD_INSTALL_MAN) $(MAN) $(MANDIR)
 
 regress:
 	scan-build $(MAKE)
